helix-term/src/commands.rs

```diff
@@ -2617,24 +2617,34 @@ fn selection_is_linewise(selection: &Selection, text: &Rope) -> bool {
     })
 }
 
-fn delete_selection_impl(cx: &mut Context, op: Operation, yank: bool) {
+enum YankMode {
+    Yank,
+    NoYank,
+}
+
+fn delete_selection_impl(cx: &mut Context, op: Operation, yank: YankMode) {
     let (view, doc) = current!(cx.editor);
 
     let selection = doc.selection(view.id);
     let only_whole_lines = selection_is_linewise(selection, doc.text());
 
-    if cx.register != Some('_') && yank {
```

Comment:
You can keep this same layout (the `if` instead of a `match`) and use the `matches!` macro here instead:

```rust
if cx.register != Some('_') && matches!(yank, YankMode::Yank) {
```